WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -161,6 +161,19 @@
       patch :update, id: registration.id, registration: { status: :accepted }
       expect(registration.reload.accepted?).to eq true
     end
+
+    it "can register for their own competition that is not yet visible" do
+      competition.update_column(:showAtAll, false)
+      competition.save!
+      expect(RegistrationsMailer).to receive(:notify_organizers_of_new_registration).and_call_original
+      expect(RegistrationsMailer).to receive(:notify_registrant_of_new_registration).and_call_original
```

Comment:
Ok I searched about this and found out that this expectation should go before the code we test because it check expect this to happen between now and the end of the spec =)